### PR TITLE
Update docs for Scheduler service going stable

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Update scheduler and jobs docs around Scheduler being stable for actor reminders and workflows.
Remove SchedulerReminders from preview features table
See if need to add to configuration schema/spec?

## Issue reference

PR will close: #4375

### CODE REVIEW

File: daprdocs/content/en/operations/support/support-preview-features.md [hunk 1]
```diff
@@ -21,5 +21,4 @@ For CLI there is no explicit opt-in, just the version that this was first made a
...
 | **Component Hot Reloading** | Allows for Dapr-loaded components to be "hot reloaded". A component spec is reloaded when it is created/updated/deleted in Kubernetes or on file when running in self-hosted mode. Ignores changes to actor state stores and workflow backends. | `HotReload`| [Hot Reloading]({{< ref components-concept.md >}}) | v1.13  |
-| **Subscription Hot Reloading** | Allows for declarative subscriptions to be "hot reloaded". A subscription is reloaded either when it is created/updated/deleted in Kubernetes, or on file in self-hosted mode. In-flight messages are unaffected when reloading. | `HotReload`| [Hot Reloading]({{< ref "subscription-methods.md#declarative-subscriptions" >}}) | v1.14  |
-| **Scheduler Actor Reminders** | Whilst the [Scheduler service]({{< ref "concepts/dapr-services/scheduler.md" >}}) is deployed by default, Scheduler actor reminders (actor reminders stored in the Scheduler control plane service as opposed to the Placement control plane service actor reminder system) are enabled through a preview feature and needs a feature flag. | `SchedulerReminders`| [Scheduler actor reminders]({{< ref "jobs-overview.md#actor-reminders" >}}) | v1.14  |
```
Comment (cicoyle) [inline daprdocs/content/en/operations/support/support-preview-features.md:25]: Should this stay in this page, but we can note its default to true? Its still how people can toggle to go back to non-scheduler reminders by setting it to false

**Review by cicoyle — APPROVED** [review]
One comment, once that is resolved lgtm

File: daprdocs/content/en/concepts/dapr-services/scheduler.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ The diagram below shows how the Scheduler service is used via the jobs API when
 
 <img src="/images/scheduler/scheduler-architecture.png" alt="Diagram showing the Scheduler control plane service and the jobs API">
 
+The Scheduler service is deployed by default, including for [Scheduler actor reminders]({{< ref "jobs-overview.md#actor-reminders" >}}) (actor reminders stored in the Scheduler control plane service as opposed to the Placement control plane service actor reminder system) and workflows.
```
Comment (msfussell) [inline daprdocs/content/en/concepts/dapr-services/scheduler.md]:
```suggestion
The Scheduler service is used by default for [actor scheduler reminders]({{< ref "jobs-overview.md#actor-reminders" >}}) where actor reminders, which are also used by workflows, are stored in the Schedule service as opposed to the Placement service.
```

File: daprdocs/content/en/developing-applications/building-blocks/jobs/jobs-overview.md [hunk 3]
```diff
@@ -61,7 +61,9 @@ The Scheduler service enables the scheduling of jobs to scale across multiple re
 
 ### Actor reminders
 
-Actors have actor reminders, but present some limitations involving scalability using the Placement service implementation. You can make reminders more scalable by using [`SchedulerReminders`]({{< ref support-preview-features.md >}}).  This is set in the configuration for your actor application. 
+Actors have actor reminders, but present some limitations involving scalability using the Placement service implementation. You can make reminders more scalable by using `SchedulerReminders` in the configuration for your actor application.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/jobs/jobs-overview.md]:
```suggestion
Actors have reminders and the Scheduler service is used by default for actor reminders (which are also used by workflows) to make them more scalable. In earlier releases the Placement service was used for actor reminders and this is no longer recommend.
```

File: daprdocs/content/en/developing-applications/building-blocks/jobs/jobs-overview.md [hunk 4]
```diff
@@ -61,7 +61,9 @@ The Scheduler service enables the scheduling of jobs to scale across multiple re
 
 ### Actor reminders
 
-Actors have actor reminders, but present some limitations involving scalability using the Placement service implementation. You can make reminders more scalable by using [`SchedulerReminders`]({{< ref support-preview-features.md >}}).  This is set in the configuration for your actor application. 
+Actors have actor reminders, but present some limitations involving scalability using the Placement service implementation. You can make reminders more scalable by using `SchedulerReminders` in the configuration for your actor application. 
+
+The `SchedulerReminders` preview feature flag defaults to `true`. To disable Scheduler actor reminders, you can manually set it to `false`.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/jobs/jobs-overview.md]:
```suggestion
The `SchedulerReminders` preview feature defaults to `true`. To disable actor scheduler reminders, you can set it to `false`.
```

File: daprdocs/content/en/operations/support/support-preview-features.md [hunk 5]
```diff
@@ -22,4 +22,4 @@ For CLI there is no explicit opt-in, just the version that this was first made a
...
+| **Scheduler Actor Reminders** | Scheduler actor reminders are actor reminders stored in the Scheduler control plane service, as opposed to the Placement control plane service actor reminder system. The `SchedulerReminders` preview feature flag defaults to `true`, but you can disable Scheduler actor reminders by manually setting it to `false`.  | `SchedulerReminders`| [Scheduler actor reminders]({{< ref "jobs-overview.md#actor-reminders" >}}) | v1.14  |
```
Comment (msfussell) [inline daprdocs/content/en/operations/support/support-preview-features.md]:
```suggestion
| **Scheduler Actor Reminders** | Scheduler actor reminders are actor reminders stored in the Scheduler control plane service, as opposed to the Placement control plane service actor reminder system. The `SchedulerReminders` preview feature flag defaults to `true`, but you can disable Scheduler actor reminders by setting it to `false`.  | `SchedulerReminders`| [Scheduler actor reminders]({{< ref "jobs-overview.md#actor-reminders" >}}) | v1.14  |
```

**Review by msfussell — COMMENTED** [review]
Some comments added. 
Given this is going to be the default there are other documents that need updating
- This Reminder data serialization format https://docs.dapr.io/developing-applications/building-blocks/actors/actors-timers-reminders/#reminder-data-serialization-format. Update the wording on the protobuf format the wording for Dapr and also say that this (presumably) only applies if you are still using the Placement service, is that correct?
- Does this partitioning actor reminders article apply to the Scheduler service or not. If not there what do we need to put about partitioning for the Scheduler service? 
- Remove the word "preview" from here https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-persisting-scheduler/ and review for any changes. Is there any addition configuration here?
- There needs to be documentation added on how to set the Scheduler service to HA mode and whether this is the default or not when deploying to Kubernetes with the init command.  In particular things like disallowing  running Scheduler with ephemeral storage when in HA mode
- Any other new end user features that need docs in v1.15?

**Comment by cicoyle** [comment]
> Some comments added. Given this is going to be the default there are other documents that need updating
> 
> * This Reminder data serialization format https://docs.dapr.io/developing-applications/building-blocks/actors/actors-timers-reminders/#reminder-data-serialization-format. Update the wording on the protobuf format the wording for Dapr and also say that this (presumably) only applies if you are still using the Placement service, is that correct?
> * Does this partitioning actor reminders article apply to the Scheduler service or not. If not there what do we need to put about partitioning for the Scheduler service?
> * Remove the word "preview" from here https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-persisting-scheduler/ and review for any changes. Is there any addition configuration here?
> * There needs to be documentation added on how to set the Scheduler service to HA mode and whether this is the default or not when deploying to Kubernetes with the init command.  In particular things like disallowing  running Scheduler with ephemeral storage when in HA mode
> * Any other new end user features that need docs in v1.15?

* Re: reminder data serialization, they will **continue** to be json serialized via the API on input/output from a user perspective. The **internal** reminders for workflow (non user facing) are protobuf serialized for performance for both scheduler + placement.
* partitioning of reminders only applies to state store reminders, not scheduler reminders. Iirc that was introduced to resolve a state store limitation initially.
* [this PR addresses the additional config](https://github.com/dapr/docs/pull/4438)
* [this PR addresses the additional config](https://github.com/dapr/docs/pull/4438) about disallowing running scheduler with ephemeral storage when in HA mode. A change that was introduced in 1.15 is that the scheduler ha now scales independent of the global ha flag. default is 1 instance, meaning ha is not default. This flag can be set via helm with `--set dapr_scheduler.ha=true`, please [read the description here for more clarification](https://github.com/dapr/dapr/pull/8304)

cc @JoshVanL for confirmation on the reminder changes bc the reminder code is currently being refactored as part of his massive PR diff that is a bit hard to grep thru the changes

**Comment by msfussell** [comment]
The HA mode for scheduler needs to be added.
1) Mention in https://docs.dapr.io/concepts/dapr-services/scheduler/#kubernetes-mode that there is a HA mode and then link to below how to do this. 
2) Can Scheduler HA mode be added with dapr init -k command? 
3) When I set global.ha.enabled=true does this also set the scheduler service to HA mode?
4) Update this topic https://docs.dapr.io/operations/hosting/kubernetes/kubernetes-production/#high-availability-mode with the information from here --set dapr_scheduler.ha=true, please https://github.com/dapr/dapr/pull/8304

File: daprdocs/content/en/concepts/dapr-services/scheduler.md [hunk 6]
```diff
@@ -11,13 +11,21 @@ The diagram below shows how the Scheduler service is used via the jobs API when
 
 <img src="/images/scheduler/scheduler-architecture.png" alt="Diagram showing the Scheduler control plane service and the jobs API">
 
+## Actor reminders
+
+Prior to Dapr v1.15, [actor reminders]({{< ref "actors-timers-reminders.md#actor-reminders" >}}) were run using the Placement service. Now, by default, the [`SchedulerReminders` feature flag]({{< ref "support-preview-features.md#current-preview-features" >}}) is set to `true`, and all new actor reminders you create are run using the Scheduler service to make them more scalable.
+
+Once you deploy Dapr v1.15, any _existing_ actor reminders are migrated from the Placement service to the Scheduler service. You can prevent this migration by setting the `SchedulerReminders` flag to `false`.
```
Comment (msfussell) [inline daprdocs/content/en/concepts/dapr-services/scheduler.md]:
```suggestion
When you deploy Dapr v1.15, any _existing_ actor reminders are migrated from the Placement service to the Scheduler service as a one time operation for each actor type. You can prevent this migration by setting the `SchedulerReminders` flag to `false` in application configuration file for the actor type.
```

**Review by msfussell — COMMENTED** [review]
LGTM. A comment

**Review by msfussell — APPROVED** [review]

**Pull request merged**